compiler/ast/ast_query.nim

```diff
@@ -662,4 +662,50 @@ proc skipAddr*(n: PNode): PNode {.inline.} =
 
 proc isNewStyleConcept*(n: PNode): bool {.inline.} =
   assert n.kind == nkTypeClassTy
-  result = n[0].kind == nkEmpty
\ No newline at end of file
+  result = n[0].kind == nkEmpty
+
+
+type
+  NamedNodePos* = enum
+    PosLastIdent ## Last item in the identifier
+    PosType ## Type positin
+    PosInit ## Initialization expression
+    PosProcBody ## Procedure body
+    PosProcReturn ## Return value of the procedure in formal parameters
+    PosProcArgs ## Formal parmeters in the procedure
+    PosTypeBody ## Type definition body
+    PosName ## Procedure name
+    PosBody ## Generic statement body
+
+  NamedNodeSlice* = enum
+    SliceAllIdents ## All identifiers in the ident defs
+    SliceAllArguments  ## All arguments in the formal parammeters
+    SliceAllBranches ## All case statement branches
+    SliceBranchExpressions ## All expressions in the `of` branch
+
+func `[]`*(node: PNode, pos: NamedNodePos): PNode =
```

Comment:
I think this should dispatch on n.kind then process the pos?

So PosType can be the return type for a proc (not a silly default) and for var let const it's the second last.

Also for procs the last one isn't always the body iirc, I think we put things on the end in certain cases getting a position 7 and 8. Can't recall the circumstances